### PR TITLE
Adding tests for disk, user, userpolicy, group & pool

### PR DESCRIPTION
**Pull request by kousalya-hpe**
Signed-off-by: Kousalya Nachimuthu <kousalya.nachimuthu@hpe.com>

### CODE REVIEW

File: test/disk_wf_test.go
Comment (prachiruparelia-hpe) [inline test/disk_wf_test.go]: This should probably be SetupSuite().

File: test/group_pool_wf_test.go
Comment (prachiruparelia-hpe) [inline test/group_pool_wf_test.go]: Are we waiting 1 min or 2 minutes here?

File: test/disk_wf_test.go
Comment (kousalya-hpe) [reply test/disk_wf_test.go]: changed

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: 2 mins, updated the comment.

File: test/disk_wf_test.go
Comment (rkumpf) [inline test/disk_wf_test.go]: Add an assertion that diskID is not nil.

File: test/disk_wf_test.go
Comment (rkumpf) [inline test/disk_wf_test.go]: Does this cause any potential harm?  Does one need to be careful about which array is used as a target for these tests?

File: test/disk_wf_test.go
Comment (rkumpf) [inline test/disk_wf_test.go]: "User Role"?

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: Why is the same file used for both group and pool tests?

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: This doesn't need to be part of the suite.

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: Ideally the tests could be written without long delays.  I wonder if it is sufficient to check that the group merge has started and not bother waiting for it to finish.

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: This test requires setup of multiple arrays, and it is destructive.  It seems like a lot to require just for testing the SDK.  Will there be an option to skip destructive tests?  Where will be the instructions for setting up and running the tests (and the actions required afterward if the tests are to be re-run)?

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: I think this just gets the first character from the version string.  It won't work if the major version reaches 10.

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: If we just use the first character of the version string (at line 115), version 5.2 would be considered "5" and would be less than 5.1.

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: Even the "default" pool can be renamed.  Is it a prerequisite for this test to have a pool named "default"?

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: Typo *existing

File: test/group_pool_wf_test.go
Comment (rkumpf) [inline test/group_pool_wf_test.go]: This seems like another notable prerequisite.

File: test/user_wf_test.go
Comment (rkumpf) [inline test/user_wf_test.go]: What should happen if userObj is nil?

File: test/user_wf_test.go
Comment (rkumpf) [inline test/user_wf_test.go]: This seems like a generally useful function.  Should it be in a common place?

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: For group & pool, I need 2 arrays to perform these tests like group merge & pool creation. So combined both the tests in same file as I will be having an additional array after group merge.

File: test/disk_wf_test.go
Comment (kousalya-hpe) [reply test/disk_wf_test.go]: my mistake, I'll change the message

File: test/disk_wf_test.go
Comment (kousalya-hpe) [reply test/disk_wf_test.go]: There won't be any harm in doing this test. We will take one 'in-use' disk slot, remove it first and add it back.

File: test/user_wf_test.go
Comment (kousalya-hpe) [reply test/user_wf_test.go]: will add an assertion for userObj nil.

File: test/disk_wf_test.go
Comment (vidhutsingh) [reply test/disk_wf_test.go]: Add assertion

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: Yes, assuming that the 'default' pool is present, so we could merge pool to default pool.

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: Since these tests are added with group merge tests, we will also be performing pool merge & unassign one array from the pool. There will be a free array(unassignedArrayName) for new pool creation. So this is not a prerequisite.

File: test/group_pool_wf_test.go
Comment (vidhutsingh) [reply test/group_pool_wf_test.go]: seems ok .

File: test/group_pool_wf_test.go
Comment (vidhutsingh) [reply test/group_pool_wf_test.go]: Split will not work properly if currentVersion is 5.x . It will just get 5 . pls fix it .

File: test/group_pool_wf_test.go
Comment (vidhutsingh) [reply test/group_pool_wf_test.go]: should be okay since by default new arrays contains "default" pool .

**Review by vidhutsingh — CHANGES_REQUESTED** [review]
please address review comments .

**Comment by vidhutsingh** [comment]
Cancelling this PR and will create new PR for containervision unit test support

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: will remove "sourcearrayGroupService"

File: test/user_wf_test.go
Comment (kousalya-hpe) [reply test/user_wf_test.go]: There is already a function for array connectivity in common, but we cannot parse arguments to it. I'll see how we can merge these two functions together and will make the changes in next PR.

File: test/disk_wf_test.go
Comment (kousalya-hpe) [reply test/disk_wf_test.go]: added assertion for diskID

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: corrected in code.

File: test/group_pool_wf_test.go
Comment (kousalya-hpe) [reply test/group_pool_wf_test.go]: Makes sense. Later we would be adding tests for replication as-well, which will require multi arrays. We'll come-up with some instructions &  an option which would skip running such tests. For now we can keep the name as 'group_pool_wf_xtest.go', so it will skip this test file.

**Review by vidhutsingh — APPROVED** [review]

**Comment by kousalya-hpe** [comment]
@rkumpf are there any other comments for this PR, can I merge it ?

**Comment by rkumpf** [comment]
I won't add further comments.  You have approval to merge.

**Pull request merged**
